docs/migration_guides/0-5-migration.md

```diff
@@ -0,0 +1,281 @@
+polk and wardell: i need a pickup of a bagels from Bakers Co at 331 5th street at 11:00am on june 3 with a dropoff at 75th Ave at 5:30pm same day
+operator: 331 5th street bagels quote - $34.50
+polk and wardell: accepted
+"""
+
+prompt = """
+From the chat exchanges below extract a schedule of deliveries.
+Chats:
+${chat_history}
+"""
+
+tools = [] # an open ai compatible list of tools
+
+response = guard(
+    openai.chat.completions.create,
+    model="gpt-4o",
+    instructions="You are a helpful assistant.",
+    prompt=prompt,
+    prompt_params={"chat_history": chat_history},
+    tools=guard.add_json_function_calling_tool(tools),
+    tool_choice="required",
+)
+```
+
+### `Guard.use()` now works for all Guards
+
+Previously, constructing a Guard via the `use` method was only supported for unstructured response schemas.  It now supports specifying validators for any Guard regardless of the initialization method (`Guard()`, `Guard.from_rail()`, `Guard.from_pydantic()`, etc.).  `Guard.use()` is also the new method of applying input validations to a Guard.
+
+Example of applying input validation to the Prompt:
+```py
+import openai
+from guardrails import Guard
+from guardrails.errors import ValidationError
+from guardrails.hub import DetectPII
+from guardrails.types import OnFailAction
+
+guard = Guard()
+guard.use(
+    DetectPII(
+        pii_entities=["EMAIL_ADDRESS", "PHONE_NUMBER"], 
+        on_fail=OnFailAction.EXCEPTION
+    ), 
+    on="prompt"
+)
+
+try:
+    guard(
+        openai.chat.completions.create,
+        prompt="My email address is not_a_real_email@guardrailsai.com",
+    )
+except ValidationError as e:
+    print(e)
+```
+
+To utilize `Guard.use()` on a Guard with structured output, you can specify a JSON Path to identify which property the Validator(s) should be assigned to.
+
+Example:
+```py
+import json
+from pydantic import BaseModel, Field
+from guardrails import Guard, OnFailAction
+from guardrails.errors import ValidationError
+from guardrails.hub import RegexMatch, ValidRange
+
+class Person(BaseModel):
+    name: str
+    # Existing way of assigning validators; still valid
+    age: int = Field(validators=[ValidRange(0, 100, on_fail=OnFailAction.EXCEPTION)])
+    is_employed: bool
+
+guard = Guard.from_pydantic(Person)
+
+# Use a regex to make sure the name is Title Case
+guard.use(
+    RegexMatch("^(?:[A-Z][^\\s]*\\s?)+$", on_fail=OnFailAction.EXCEPTION),
+    on="$.name"
+)
+
+try:
+    guard.validate(json.dumps({
+        "name": "john doe",
+        "age": 30,
+        "is_employed": True
+    }))
+except ValidationError as e:
+    print(e)
+```
+
+## Backwards-incompatible changes
+
+### Args vs Kwargs
+In previous versions, most of the Guardrails interfaces utilized positional arguments for most parameters. This could be tedious when specifying optional arguments.
+
+In 0.5.0, for our public interfaces, only required arguments are positional; all optional arguments are key-word only.
```

Comment:
Low priority nit: `keyword-only` or `keyword only`.  I'm not sure if keyword should be split here.

Reply:
Good catch, I think my brain was stuck on `key-value`.